workbench-core/example/infrastructure/integration-tests/tests/datasets/accessPermissions.test.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { dataSetPrefix } from '@aws/workbench-core-example-app/lib/configs/constants';
+import { CreateUser } from '@aws/workbench-core-user-management';
+import { v4 as uuidv4 } from 'uuid';
+import ClientSession from '../../support/clientSession';
+import Dataset from '../../support/resources/datasets/dataset';
+import Setup from '../../support/setup';
+import HttpError from '../../support/utils/HttpError';
+
+describe('DataSets access permissions integration tests', () => {
+  const setup: Setup = new Setup();
+  const mockBadValue: string = 'fake-data';
+  let adminSession: ClientSession;
+  let user: CreateUser;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+    user = {
+      firstName: 'Test',
+      lastName: 'User',
+      email: `success+create-user-${uuidv4()}@simulator.amazonses.com`
+    };
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('addDataSetAccessPermissions', () => {
+    it('Adds a read-only access permission.', async () => {
+      let response = await adminSession.resources.datasets.create({}, true);
+      const dataSetId: string = response.data.id;
+      response = await adminSession.resources.groups.create({}, true);
+      const { groupId } = response.data;
+      await expect(
+        (adminSession.resources.datasets.children.get(dataSetId) as Dataset).addAccess({
+          permission: {
+            identityType: 'GROUP',
+            identity: groupId,
+            accessLevel: 'read-only'
+          }
+        })
+      ).resolves.toMatchObject({
+        data: {
+          dataSetId: dataSetId,
+          permissions: [
+            {
+              identityType: 'GROUP',
+              identity: groupId,
+              accessLevel: 'read-only'
+            }
+          ]
+        }
+      });
+    });
+    it('Adds a read-only and a read-write access permission when read-write is requested.', async () => {
```

Comment:
This confuses me a little bit: how can you have `read-only` AND `read-write` permissions on the same dataset? They're conflicting permissions, right?

Reply:
I can change this behavior. 

read-only from an authz perspective is simply `READ`.
read-write is `READ` and `UPDATE` (and if you need them, probably `CREATE`/`DELETE`, but not needed at this juncture for our purposes).

I can create a private method to manage the conversion from READ to read-only and READ/UPDATE to read-write. I originally just used `UPDATE` to capture read-write, but somebody else pushed back on that.

Reply:
okay. finally! I ended up doing some future proofing and separated out converting the dynamic auth output into datasets obejcts into a private method which needed a bunch of unit testing of its own. It's a bit more robust than what I had too.

But now, the READ permission on read-write access will be ignored and UPDATE will supersede with the read-write response.